packages/premint-sdk/src/premint-api.ts

```diff
@@ -0,0 +1,684 @@
+};
+
+const OPEN_EDITION_MINT_SIZE = "18446744073709551615";
+export const DefaultMintArguments = {
+  maxSupply: BigInt(OPEN_EDITION_MINT_SIZE),
+  maxTokensPerAddress: 0n,
+  pricePerToken: 0n,
+  mintDuration: BigInt(60 * 60 * 24 * 7), // 1 week
+  mintStart: 0n,
+  royaltyMintSchedule: 0,
+  royaltyBPS: 1000, // 10%,
+};
+
+function getLogFromReceipt(receipt: TransactionReceipt) {
+  for (const data of receipt.logs) {
+    try {
+      const decodedLog = decodeEventLog({
+        abi: zoraCreator1155PremintExecutorImplABI,
+        eventName: "Preminted",
+        ...data,
+      });
+      if (decodedLog.eventName === "Preminted") {
+        return decodedLog.args;
+      }
+    } catch (err: any) {}
+  }
+}
+
+/**
+ * Premint server response type.
+ */
+export type PremintResponse = {
+  collection: {
+    contractAdmin: Address;
+    contractURI: string;
+    contractName: string;
+  };
+  premint: {
+    tokenConfig: {
+      tokenURI: string;
+      maxSupply: string;
+      maxTokensPerAddress: string;
+      pricePerToken: string;
+      mintStart: string;
+      mintDuration: string;
+      royaltyMintSchedule: number;
+      royaltyBPS: number;
+      royaltyRecipient: Address;
+      fixedPriceMinter: Address;
+    };
+    uid: number;
+    version: number;
+    deleted: boolean;
+  };
+  chain_name: BackendChainNames;
+  signature: Hex;
+};
+
+/**
+ * Convert server to on-chain types for a premint
+ * @param premint Premint object from the server to convert to one that's compatible with viem
+ * @returns Viem type-compatible premint object
+ */
+export const convertPremint = (premint: PremintResponse["premint"]) => ({
+  ...premint,
+  tokenConfig: {
+    ...premint.tokenConfig,
+    maxSupply: BigInt(premint.tokenConfig.maxSupply),
+    pricePerToken: BigInt(premint.tokenConfig.pricePerToken),
+    mintStart: BigInt(premint.tokenConfig.mintStart),
+    mintDuration: BigInt(premint.tokenConfig.mintDuration),
+    maxTokensPerAddress: BigInt(premint.tokenConfig.maxTokensPerAddress),
+  },
+});
+
+/**
+ * Convert on-chain types for a premint to a server safe type
+ * @param premint Premint object from viem to convert to a JSON compatible type.
+ * @returns JSON compatible premint
+ */
+export const encodePremintForAPI = ({
+  tokenConfig,
+  ...premint
+}: PremintConfig) => ({
+  ...premint,
+  tokenConfig: {
+    ...tokenConfig,
+    maxSupply: tokenConfig.maxSupply.toString(),
+    pricePerToken: tokenConfig.pricePerToken.toString(),
+    mintStart: tokenConfig.mintStart.toString(),
+    mintDuration: tokenConfig.mintDuration.toString(),
+    maxTokensPerAddress: tokenConfig.maxTokensPerAddress.toString(),
+  },
+});
+
+/**
+ * Zora API Server base URL
+ */
+const ZORA_PREMINT_API_BASE = "https://api.zora.co/premint/";
+
+/**
+ * Preminter API to access ZORA Premint functionality.
+ * Currently only supports V1 premints.
+ */
+export class PremintAPI {
+  network: NetworkConfig;
+  chain: Chain;
+  rewardPerToken: bigint;
```

Comment:
this should be a static constant value that can be defined outside of the class since it doesn't vary per instance of the class